desktop/dev-environments.md

```diff
@@ -253,6 +253,24 @@ In this preview, Dev Environments support a JSON file which allows you to specif
 }
 ```
 
+Next, you have to define the dependencies you want to include in your `Dockerfile.devenv`, alongside the following requisites:
+
+While some images or Dockerfiles will include a non-root user, many base images and Dockerfiles do not. Fortunately, you can add a non-root user named `vscode`. If you were to include the Docker tooling (e.g. `docker` cli, `docker compose`, etc.) in the `Dockerfile.devenv`, you would need the `vscode` user to be included in the `docker` group. Finally, remember to set `sleep infinity` in the entrypoint to keep the dev container running indefinitely.
```

Comment:
> Finally, remember to set `sleep infinity` in the entrypoint to keep the dev container running indefinitely.

Perhaps a bit off-topic (just now noticing this part); setting `sleep infinity` in entrypoint seems like a bit of a large hammer (and makes the image/container harder to use).

IIUC, we (can) control how the container is started, correct? I'm wondering if instead we could just make the container run with different options to "make it work".

When trying to run a container "detached" but the container's process is not a long-living process (e.g. has `/bin/sh` or `/bin/bash` as entrypoint), starting it detached will make it exit immediately; this is because the shell detects there's no terminal attached, so it has nothing to do;

```bash
docker run -d busybox
eb94e6e28b41bc5bb6f651cc54e3aab5a46a56d805f3787eeb9451a54301af47

docker ps -a

CONTAINER ID   IMAGE                        COMMAND                  CREATED          STATUS                      PORTS                            NAMES
eb94e6e28b41   busybox                      "sh"                     15 seconds ago   Exited (0) 12 seconds ago                                    serene_keldysh
```

However, starting the same image with a TTY attached (`-t` or `-it`) will keep it running; the shell detects that a TTY is attached, so keeps waiting; here's a busybox container that's started with both `-d` (detached) and `-t` (a tty);

```bash
docker run -dt busybox
5b7e989747625b023144a7c5ab0af092ea0708f13bcdc58994bdb591d7a19e22

docker ps -a
CONTAINER ID   IMAGE                        COMMAND                  CREATED              STATUS                          PORTS                            NAMES
5b7e98974762   busybox                      "sh"                     3 seconds ago        Up 2 seconds                                                     nice_black
eb94e6e28b41   busybox                      "sh"                     About a minute ago   Exited (0) About a minute ago                                    serene_keldysh
```

Reply:
We can remove the sentence about setting the entrypoint to `sleep infinity` to not confuse the user, the Dev Env backend server is taking care of this.